tests/Unit/Domain/Test_OrientationMap.cpp

```diff
@@ -261,6 +262,203 @@ SPECTRE_TEST_CASE("Unit.Domain.OrientationMap", "[Domain][Unit]") {
   test_3d();
 }
 
+// [[OutputRegex, This OrientationMap fails to map Directions one-to-one.]]
+[[noreturn]] SPECTRE_TEST_CASE("Unit.Domain.OrientationMap.Bijective",
+                               "[Domain][Unit]") {
+  ASSERTION_TEST();
+#ifdef SPECTRE_DEBUG
+  auto failed_orientationmap = OrientationMap<2>{std::array<Direction<2>, 2>{
+      {Direction<2>::upper_xi(), Direction<2>::lower_xi()}}};
+  static_cast<void>(failed_orientationmap);
+
+  ERROR("Failed to trigger ASSERT in an assertion test");
+#endif
+}
+
+// [[OutputRegex, This OrientationMap fails to map Directions one-to-one.]]
+[[noreturn]] SPECTRE_TEST_CASE("Unit.Domain.OrientationMap.BijectiveHost",
+                               "[Domain][Unit]") {
+  ASSERTION_TEST();
+#ifdef SPECTRE_DEBUG
+  auto failed_orientationmap = OrientationMap<2>{
+      std::array<Direction<2>, 2>{
+          {Direction<2>::upper_xi(), Direction<2>::lower_xi()}},
+      std::array<Direction<2>, 2>{
+          {Direction<2>::upper_xi(), Direction<2>::upper_eta()}},
+  };
+  static_cast<void>(failed_orientationmap);
+
+  ERROR("Failed to trigger ASSERT in an assertion test");
+#endif
+}
+
+// [[OutputRegex, This OrientationMap fails to map Directions one-to-one.]]
+[[noreturn]] SPECTRE_TEST_CASE("Unit.Domain.OrientationMap.BijectiveNeighbor",
+                               "[Domain][Unit]") {
+  ASSERTION_TEST();
+#ifdef SPECTRE_DEBUG
+  auto failed_orientationmap = OrientationMap<3>{
+      std::array<Direction<3>, 3>{{Direction<3>::upper_xi(),
+                                   Direction<3>::lower_eta(),
+                                   Direction<3>::lower_zeta()}},
+      std::array<Direction<3>, 3>{{Direction<3>::upper_xi(),
+                                   Direction<3>::upper_eta(),
+                                   Direction<3>::lower_eta()}},
+  };
+  static_cast<void>(failed_orientationmap);
+
+  ERROR("Failed to trigger ASSERT in an assertion test");
+#endif
+}
+
+namespace {
+template <size_t VolumeDim>
+OrientationMap<VolumeDim> generate_orientationmap(
+    size_t num_of_cycle_permutation, size_t num_of_sign_permutation) {
+  std::array<size_t, VolumeDim> dimensions{};
+  std::bitset<VolumeDim> bitset = num_of_sign_permutation;
+  std::array<Direction<VolumeDim>, VolumeDim> array_of_directions{};
+  for (size_t i = 0; i < VolumeDim; i++) {
+    gsl::at(dimensions, i) = i;
+  }
+  for (size_t i = 0; i < num_of_cycle_permutation; i++) {
+    std::next_permutation(dimensions.begin(), dimensions.end());
+  }
+  for (size_t i = 0; i < VolumeDim; i++) {
+    gsl::at(array_of_directions, i) = Direction<VolumeDim>{
+        gsl::at(dimensions, i), bitset[i] == 1 ? Side::Upper : Side::Lower};
+  }
+  return OrientationMap<VolumeDim>{array_of_directions};
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.DiscreteRotation.AllOrientations",
```

Comment:
The discrete rotations part of this commit should be in the DiscreteRotation commit